### PR TITLE
Wrap the apicurio-client-gen distribution mechanism

### PR DESCRIPTION
**Pull request by andreaTP**

### CODE REVIEW

File: ui/webpack.common.js [hunk 1]
```diff
@@ -91,16 +90,11 @@ module.exports = (env, argv) => {
...
-            'echo Unzipping client-gen-' + clientGenVersion + '.zip...',
-            'unzip -q resources/client-gen-' + clientGenVersion + '.zip -d client-gen',
-            'echo Done!'
-          ],
+          scripts: [ 'resources/download-client-gen-dist.sh' ],
           blocking: true,
           parallel: false
-        }
+        },
+        safe: true,
```
Comment (andreaTP) [inline ui/webpack.common.js:97]: this prevents silent failures of the script.

File: ui/resources/download-client-gen-dist.sh [hunk 2]
```diff
@@ -1,17 +1,62 @@
...
+# - `APICURIO_CLIENT_GEN_FRESH` -> forcefully download from remote the distribution archive
 
 SCRIPT_DIR=$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &>/dev/null && pwd)
 
-OUTPUT_FILE="$SCRIPT_DIR/client-gen-$VERSION.zip"
+VERSION="${APICURIO_CLIENT_GEN_VERSION-"0.1.2"}"
+echo "Using version: ${VERSION}"
+
+UPSTREAM_DIST_URL="https://github.com/Apicurio/apicurio-client-gen/releases/download/${VERSION}/dist.tar.gz"
+PRODUCT_DIST_URL="http://download.eng.bos.redhat.com/rcm-guest/staging/integration/service-registry-dev/dist/apicurio-client-gen-dist-${VERSION}.tar.gz"
```
[the remark this answers is not in the document]
Comment (riprasad) [reply ui/resources/download-client-gen-dist.sh:17]: it's behind vpn and it's not a production server, so i think it should be fine

[the remark this answers is not in the document]
Comment (riprasad) [reply ui/resources/download-client-gen-dist.sh:17]: but I agree it's a bad practice. maybe. We will set the actual URL in our product build configs.

[the remark this answers is not in the document]
Comment (andreaTP) [reply ui/resources/download-client-gen-dist.sh:17]: the server is not accessible outside the VPN, but, as you prefer, I can remove it 👍

[the remark this answers is not in the document]
Comment (andreaTP) [reply ui/resources/download-client-gen-dist.sh]: done

**Review by riprasad — APPROVED** [review]
LGTM

**Review by EricWittmann — APPROVED** [review]

**Pull request merged**